### PR TITLE
Split developer guide up into multiple pages

### PR DESCRIPTION
**Pull request by Cobra-Bitcoin**
Same as https://github.com/bitcoin-dot-org/bitcoin.org/pull/2700 but fixes the issue raised in https://github.com/bitcoin-dot-org/bitcoin.org/issues/2739.

Links like "developer-guide#term-target" will get redirected to their equivalent within each guide page such as "blockchain-guide#term-target".

### CODE REVIEW

**Comment by harding** [comment]
@Cobra-Bitcoin I'd suggest that you also (here or in a separate PR that this PR is based upon) revert https://github.com/bitcoin-dot-org/bitcoin.org/commit/2b7882952abd63e777627fb889fe1b8abdf7a8c9 as described in #2739, as that test can assure you through Travis CI that none of the internal anchor links are broken, as was the case before.

**Comment by Cobra-Bitcoin** [comment]
@harding Thanks! I've opened a separate PR for that too: https://github.com/bitcoin-dot-org/bitcoin.org/pull/2786.

**Comment by Cobra-Bitcoin** [comment]
I think this is ready to be merged soon, any objections? @wbnns

**Comment by wbnns** [comment]
@Cobra-Bitcoin Ok, thanks. Will check alongside #2786.

**Comment by wbnns** [comment]
Quick update on this PR - during additional review, it appears the navigation can become broken when viewing the site in a language other than English. The links are OK on the main dev doc page, but they redirect to `/en`/ on click. If the language is then changed, page links become broken:

image

For example, navigating to `/uk/p2p-network-guide` after changing the language to `uk` like in the above screenshot, would result in a 404. This is because new dev docs subpages are only generated for the `en` locale. We could potentially add redirects to workaround this, however, dev docs currently link to the glossary in other locales. If we redirect, this linkage would be lost.

**Comment by Cobra-Bitcoin** [comment]
@wbnns Good find, will fix shortly!

**Comment by wbnns** [comment]
@Cobra-Bitcoin Ok, thanks!

**Comment by 3ggerman29** [comment]
my btc  Wallet address 1Hk3mhTPRvsv5RCbMWZqMWJmt5Tk8NxQFF

**Comment by Cobra-Bitcoin** [comment]
Fixed. Ignore the failing tests. The anchor links are being redirected through Javascript and the recently merged test (https://github.com/bitcoin-dot-org/bitcoin.org/pull/2786) to detect broken links isn't capable of detecting Javascript redirects.

change

It should be impossible to navigate to links like /uk/p2p-network-guide now because the developer guide page won't link to it, it'll always link to the English version of the page, despite the language currently selected, as shown in the screenshot where the language is set to Ukrainian. The screenshot was taken while hovering over the link to the P2P Network Guide page.

**Comment by wbnns** [comment]
@Cobra-Bitcoin 

Apologies, I don't see the new commit which resolves the issue with JS in association with the last [comment](https://github.com/bitcoin-dot-org/bitcoin.org/pull/2780#issuecomment-480522368) - I only see the merge commit (f842a03). Is there a new commit corresponding to the fix?

**Comment by Cobra-Bitcoin** [comment]
I pushed an update to fix the broken links. All tests are passing now.

I had to rebase and I think that did something to fix the issue. When I mentioned JS, I was referring to redirecting all the old broken link redirects like "developer-guide#term-target" ->  "blockchain-guide#term-target", which the link test checker wasn't detecting. Now it's irrelevant though, since I manually fixed all the links myself to point to their respective locations, though the JS redirects are still a necessity to keep because of other sites still linking to the old links.

If you test again, you should see it's impossible to get linked to something like `/uk/p2p-network-guide` presently because nowhere on the site links to that page, and the tests would fail if it did.

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ default: clean build
 
 ## `make preview`: start the built-in Jekyll preview
 preview: clean
-	$S export LANG=C.UTF-8 ; bundle exec jekyll serve --incremental
+	$S export LANG=C.UTF-8 ; bundle exec jekyll serve --incremental --host=0.0.0.0
```
Comment (wbnns) [inline Makefile]: We can discard change, assuming it's an artifact left over from testing.

Comment (Cobra-Bitcoin) [reply Makefile]: Thanks, I've pushed an update discarding the change.

**Comment by Cobra-Bitcoin** [comment]
@wbnns Fixed.

**Comment by wbnns** [comment]
@Cobra-Bitcoin Ok, great, thanks. Will test again. 👍

**Comment by wbnns** [comment]
This is scheduled to be merged on Monday, May 6th.

**Pull request merged**